xdsl/transforms/experimental/dmp/stencil_global_to_local.py

```diff
@@ -374,79 +373,29 @@ def else_() -> Iterable[Operation]:
 
 
 def generate_memcpy(
-    source: SSAValue, ex: dmp.HaloExchangeDecl, dest: SSAValue, reverse: bool = False
+    field: SSAValue, ex: dmp.HaloExchangeDecl, buffer: SSAValue, receive: bool = False
 ) -> list[Operation]:
     """
     This function generates a memcpy routine to copy over the parts
-    specified by the `ex` from `source` into `dest`.
+    specified by the `field` from `field` into `buffer`.
 
-    If reverse=True, it instead copy from `dest` into the parts of
-    `source` as specified by `ex`
+    If receive=True, it instead copy from `buffer` into the parts of
+    `field` as specified by `ex`
 
     """
-    assert ex.dim == 2, "Cannot handle non-2d case of memcpy yet!"
-
-    idx = builtin.IndexType()
-
-    x0 = arith.Constant.from_int_and_width(ex.offset[0], idx)
-    x0.result.name_hint = "x0"
-    y0 = arith.Constant.from_int_and_width(ex.offset[1], idx)
-    y0.result.name_hint = "y0"
-    x_len = arith.Constant.from_int_and_width(ex.size[0], idx)
-    x_len.result.name_hint = "x_len"
-    y_len = arith.Constant.from_int_and_width(ex.size[1], idx)
-    y_len.result.name_hint = "y_len"
-    cst0 = arith.Constant.from_int_and_width(0, idx)
-    cst1 = arith.Constant.from_int_and_width(1, idx)
-
-    @Builder.implicit_region([idx, idx])
-    def loop_body(args: tuple[BlockArgument, ...]):
-        """
-        Loop body of the scf.parallel() that iterates the following domain:
-        i = (0->x_len), y = (0->y_len)
-        """
-        i, j = args
-        i.name_hint = "i"
-        j.name_hint = "j"
-
-        # x = i + x0
-        # y = i + y0
-        x = arith.Addi(i, x0)
-        y = arith.Addi(j, y0)
-
-        x.result.name_hint = "x"
-        y.result.name_hint = "y"
-
-        # linearized_idx = (j * x_len) + i
-        dest_idx = arith.Muli(j, x_len)
-        linearized_idx = arith.Addi(dest_idx, i)
-        linearized_idx.result.name_hint = "linearized_idx"
-
-        if reverse:
-            load = memref.Load.get(dest, [linearized_idx])
-            memref.Store.get(load, source, [x, y])
-        else:
-            load = memref.Load.get(source, [x, y])
```

Comment:
So all thsese are just dropped now? great

Reply:
Yup, that's `memref.memcopy`'s job :slightly_smiling_face: